### PR TITLE
PHP-1365: Validate cursor command option, but don't enforce batchSize

### PR DESCRIPTION
**Pull request by jmikola**
https://jira.mongodb.org/browse/PHP-1365

----

The driver should not enforce the cursor.batchSize command option, as the server will already use its own default. We should validate that the cursor option is an array/object, and empty arrays should be converted to an object to ensure proper BSON serialization (the server requires an object type).

Only aggregateCursor() requires that we inject the cursor option if it is missing. Collection and index enumeration commands do not require the option at all, as they will always return cursors in 2.8.0-RC4+ and return inline results in earlier versions. RC3 can return an emulated cursor response with only a first batch, but it is practically equivalent to the result when the cursor option is not specified at all.

### CODE REVIEW

**Comment by jmikola** [comment]
Conveniently, `tests/generic/mongocollection-aggregatecursor-005.phpt` already tests for passing `[ 'cursor' => [] ]` as a command option, so we did not need a new test to verify that empty arrays are converted to objects.

File: tests/generic/mongocollection-aggregatecursor-007.phpt [hunk 1]
```diff
@@ -38,6 +39,7 @@ for ($i = 0; $i < 103; $i++) {
     $collection->insert(array('article_id' => $i));
 }
 
+// This assumes that the server's batch size defaults to 101 for aggregate
```
Comment (derickr) [inline tests/generic/mongocollection-aggregatecursor-007.phpt]: Do we still want to rely on that, or perhaps also have a test where we force it to 101? The main thing here is that we make sure we test that getMore works.

File: command_cursor.c [hunk 2]
```diff
@@ -105,28 +107,38 @@ int php_mongo_enforce_batch_size_on_command(zval *command, int size TSRMLS_DC)
...
-	if (Z_TYPE_P(zcursor) != IS_ARRAY && Z_TYPE_P(zcursor) != IS_OBJECT) {
+	if (Z_TYPE_PP(zcursor) != IS_ARRAY && Z_TYPE_PP(zcursor) != IS_OBJECT) {
 		php_mongo_cursor_throw(mongo_ce_CursorException, NULL, 32 TSRMLS_CC, "The cursor command's 'cursor' element is not an array or object");
 		return 0;
 	}
 
-	if (!zend_hash_exists(HASH_P(zcursor), "batchSize", 10)) {
-		zval *tmp_batchsize;
+	if (Z_TYPE_PP(zcursor) == IS_ARRAY && zend_hash_num_elements(Z_ARRVAL_PP(zcursor)) == 0) {
+		convert_to_object(*zcursor);
```
Comment (derickr) [inline command_cursor.c]: I never knew that one existed :-)

File: tests/generic/mongocollection-aggregatecursor-007.phpt (hunk 1 above)
Comment (jmikola) [inline tests/generic/mongocollection-aggregatecursor-007.phpt]: In `tests/generic`, tests 006 and 008 for `aggregateCursor()` do test getmore requests with explicit batch sizes of 0 and 42, respectively.

**Comment by derickr** [comment]
LGTM
